src/utils/FrameConfigUpdateQueue.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-FileCopyrightText: 2025 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.24;
+
+import { IReportAsyncProcessor } from "../lib/base-oracle/interfaces/IReportAsyncProcessor.sol";
+import { IConsensusContract } from "../lib/base-oracle/interfaces/IConsensusContract.sol";
+
+/// @notice Helper to perform sequential updates of the frame configuration
+///        in the HashConsensus contract used by the CSFeeOracle.
+///        It helps to safely change the frame configuration on expected oracle report,
+///        ensuring that there is no missing frame(s) during the update.
```

Comment:
The main idea of the contract is to offset report windows. While it is great to keep the description general, I suggest explicitly pointing out the main use case assumed.